katdal/dataset.py

```diff
@@ -88,21 +74,34 @@ def __init__(self, ants, corr_products):
     def __repr__(self):
         """Short human-friendly string representation of subarray object."""
         return "<katdal.Subarray antennas=%d inputs=%d corrprods=%d at 0x%x>" % \
-               (len(self.ants), len(self.inputs), len(self.corr_products), id(self))
+               (len(self.ants), len(self.inputs), len(self.corr_products),
+                id(self))
+
+    @property
+    def _description(self):
+        """Complete string representation, used internally for comparisons."""
+        ants = '\n'.join(ant.description for ant in self.ants)
+        corrprods = ' '.join('%s,%s' % (inpA, inpB)
+                             for inpA, inpB in self.corr_products)
+        return '\n'.join((ants, corrprods))
```

Comment:
You're no longer comparing `inputs`. Could that lead to incorrect comparisons if two Subarrays have the same antennas and corrprods but the inputs are listed in a different order?